### ISSUE TITLE
Were you able to build?

### ISSUE THREAD

**Issue by rlingineni** [elliotberry/wacom-inkspace#1]
Hey,

This was really cool - didn't know this was possible.

I was trying to build this code with `npm run start` so I could understand how the components fit together and inspect the elements - but it didn't work for me. Are there any pre-requisites to building from source? 

I tried an npm install - but I got some wacky error related to electron.

**Comment by elliotberry** [comment]
Good to have you checkin this out! I haven't been able to build this either, yet, on Mac. There's an issue with the node version related to electron-builder, but there's also a custom module that needs a home somewhere so we can re-initialize `npm install`. I'll start a new branch for development and take another look, maybe if you get anywhere you can add to it.

**Comment by elliotberry** [comment]
Ok, check out the dev branch and see if you can build. I built it successfully but on `npm run start` I get:
`> WacomInkspaceApp@2.7.0 start ~/Desktop/wacom-inkspace
> cross-env NODE_ENV=development electron .

dyld: lazy symbol binding failed: Symbol not found: __ZN2v816FunctionTemplate3NewEPNS_7IsolateEPFvRKNS_20FunctionCallbackInfoINS_5ValueEEEENS_5LocalIS4_EENSA_INS_9SignatureEEEiNS_19ConstructorBehaviorENS_14SideEffectTypeE
  Referenced from: ~/Desktop/wacom-inkspace/node_modules/leveldown/build/Release/leveldown.node
  Expected in: flat namespace

dyld: Symbol not found: __ZN2v816FunctionTemplate3NewEPNS_7IsolateEPFvRKNS_20FunctionCallbackInfoINS_5ValueEEEENS_5LocalIS4_EENSA_INS_9SignatureEEEiNS_19ConstructorBehaviorENS_14SideEffectTypeE
  Referenced from: ~/Desktop/wacom-inkspace/node_modules/leveldown/build/Release/leveldown.node
  Expected in: flat namespace`

I'll dig more into this later.

**Comment by rlingineni** [comment]
Odd - no worries. I'll check if I can build as well. 

I have a smartpad btw. What I really want to do is isolate the live sharing feature. So then I can infer the live output and create my own learning algorithm.

So - my other plan of approach is to isolate the components. I was thinking of starting at the `DeviceManager`  [code](https://github.com/elliotberry/wacom-inkspace/blob/master/src/globals/DeviceManager.js) and tracking back. 

In high level terms - I might try isolating the code to find the components that can help with these parts:

1. Connect to Device (For Slate, it seems like I can only access via BLE)
2. Get the state of the device
3. Get the transformed output of the device when live writing

**Comment by elliotberry** [comment]
I'm very much in a similar boat. My goal is to pair down the instructions to where it can be made into a simple CLI app to export drawings. I'm working on a branch where I think I've isolated the bare functions; basically, since electron is giving me the most problems, I thought it would be better to start from the CLI up. Let me add it in, I'd love thoughts and to be able to help you out.

**Comment by elliotberry** [comment]
Here it is: https://github.com/elliotberry/wacom-inkspace/tree/bare-bones.
Haven't been able to pair the device yet. The entry point should be example-device-manager.js, as this is the class for manipulating all devices in the original app. Or. potentially, a CLI/frontend could import that. Apologies, as if you're working outside OSX I've deleted the classes for windows support. I've also removed the usb support and so on for now.

It uses BLE and the Noble module in Node, which is all super easy to use. The code also includes all the of the characteristics to use on the device. All that has to be done is some basic troubleshooting, and I haven't had time yet.
